parsers/GB.py

```diff
@@ -1,71 +1,99 @@
-"""
+"""Parser that uses the RTE-FRANCE API"""
 
 import xml.etree.ElementTree as ET
-from datetime import datetime, timedelta
+from datetime import datetime, timedelta, timezone
 from logging import Logger, getLogger
+from zoneinfo import ZoneInfo
 
-import arrow
 from requests import Session
 
+from electricitymap.contrib.lib.models.event_lists import PriceList
+from electricitymap.contrib.lib.models.events import EventSourceType
+from electricitymap.contrib.lib.types import ZoneKey
 from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+PARSER = "GB.py"
+TIMEZONE = ZoneInfo("Europe/London")
+ZONE_KEY = ZoneKey("GB")
 
 
 @refetch_frequency(timedelta(days=1))
 def fetch_price(
-    zone_key: str,
+    zone_key: ZoneKey = ZONE_KEY,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list:
-    if target_datetime:
-        now = arrow.get(target_datetime, tz="Europe/Paris")
-    else:
-        now = arrow.now(tz="Europe/London")
+) -> list[dict]:
+    """Returns one-hourly prices for the requested day."""
 
-    r = session or Session()
-    formatted_from = now.shift(days=-1).format("DD/MM/YYYY")
-    formatted_to = now.format("DD/MM/YYYY")
+    now = datetime.now(timezone.utc)
+    target_datetime = (
+        now if target_datetime is None else target_datetime.astimezone(timezone.utc)
+    )
 
-    url = f"http://eco2mix.rte-france.com/curves/getDonneesMarche?dateDeb={formatted_from}&dateFin={formatted_to}&mode=NORM"
+    # API works in UTC timestamps
+    day = target_datetime.strftime("%d/%m/%Y")
+    url = f"http://eco2mix.rte-france.com/curves/getDonneesMarche?dateDeb={day}&dateFin={day}&mode=NORM"
 
-    response = r.get(url)
-    obj = ET.fromstring(response.content)
-    datas = {}
+    session = session or Session()
+    response = session.get(url)
+    if not response.ok:
+        raise ParserException(
+            PARSER,
+            f"Exception when fetching price error code: {response.status_code}: {response.text}",
+            zone_key,
+        )
 
-    for donnesMarche in obj:
-        if donnesMarche.tag != "donneesMarche":
-            continue
+    xml_tree = ET.fromstring(response.content)
 
-        start_date = arrow.get(
-            arrow.get(donnesMarche.attrib["date"]).datetime, "Europe/Paris"
+    price_list = PriceList(logger=logger)
+    for daily_market_data in xml_tree.iterfind("donneesMarche"):
+        day = datetime.strptime(daily_market_data.get("date"), "%Y-%m-%d").replace(
+            tzinfo=timezone.utc
         )
+        for daily_zone_data in daily_market_data:
+            zone_code = daily_zone_data.get("perimetre")
 
-        for item in donnesMarche:
-            if item.get("granularite") != "Global":
+            # Data for Germany / Luxembourg is not set / reported as aggregate region
+            if zone_code in {"DE", "DL"}:
                 continue
```

Comment:
note: I don't know what the best way to deal with these really is / how to interpret the API data

"DE" (germany) data is always null, and "DL" data has values. "DL" might maybe represent Germany + Luxembourg? In fact https://www.rte-france.com/eco2mix/les-donnees-de-marche displays a single aggregate `Allemagne + Lux` curve 👀  either way, I don't think they would be usable for us

Reply:
I see - thanks! I am very much not familiar with the "real life" side of things 😅 

So happy to just skip over these as done here?

Reply:
Yep I think so. But I'll double check when doing a full code review.